### PR TITLE
secret-handshake: Add canonical data (revives #159)

### PR DESCRIPTION
**Pull request by petertseng**
The time has come to help the abandoned #159 get merged.

Implementing tracks:

* https://github.com/exercism/xcsharp/blob/master/exercises/secret-handshake/SecretHandshakeTest.cs
* https://github.com/exercism/xecmascript/blob/master/exercises/secret-handshake/secret-handshake.spec.js
* https://github.com/exercism/xfsharp/blob/master/exercises/secret-handshake/SecretHandshakeTest.fs
* https://github.com/exercism/xgo/blob/master/exercises/secret-handshake/secret_handshake_test.go
* https://github.com/exercism/xhaskell/blob/master/exercises/secret-handshake/test/Tests.hs
* https://github.com/exercism/xjavascript/blob/master/exercises/secret-handshake/secret-handshake.spec.js
* https://github.com/exercism/xlua/blob/master/exercises/secret-handshake/secret-handshake_spec.lua
* https://github.com/exercism/xobjective-c/blob/master/exercises/secret-handshake/SecretHandshakeTest.m
* https://github.com/exercism/xperl5/blob/master/exercises/secret-handshake/handshake.t
* https://github.com/exercism/xpython/blob/master/exercises/secret-handshake/handshake_test.py
* https://github.com/exercism/xruby/blob/master/exercises/secret-handshake/secret_handshake_test.rb
* https://github.com/exercism/xscala/blob/master/exercises/secret-handshake/src/test/scala/SecretHandshakeTest.scala
* https://github.com/exercism/xswift/blob/master/exercises/secret-handshake/SecretHandshakeTest.swift

Summary of tests among these languages:

* All language tracks test: 1, 2, 4, 8, 3, 19, 31.
* Go, Haskell, Objective C, Scala, and Swift additionally test 0.
* Go is the only track that tests 32, 33.
* Python is the only track that tests 9, 22, 5, -9.
* Python is the only track that tests action -> number conversions.
* Haskell, Python, and Scala test strings (string binary -> number conversion)
* ECMAScript and JavaScript test strings (only that strings are rejected)
* Perl and Ruby test strings (that they are accepted but result in an empty array)

Closes #159 by way of replacement.
Closes exercism/todo#145

### CODE REVIEW

File: exercises/secret-handshake/canonical-data.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                "input": 19,
+                "expected": ["double blink", "wink"]
+            },
+            {
+                "description": "reversing one action gives the same action",
+                "input": 24,
+                "expected": ["jump"]
+            },
+            {
+                "description": "reversing no actions still gives the no actions",
```
Comment (petertseng) [inline exercises/secret-handshake/canonical-data.json]: the no?

**Comment by petertseng** [comment]
I have to not use the term "reapply" since I have in the past used it to mean "revert a revert commit", so I don't want to overload the term.

**Comment by Insti** [comment]
Even though I've approved the current version, I also agree with @IanWhitney 's points.

**Comment by petertseng** [comment]
I've a decision to make.

Situation:

* All tracks test 3 and 19.
* README has 9 and "11001" (25)

I could do one of two things:

* Make README use 3 and 19
* Add 9 and 25 to the test file, in addition to 3 and 19 (but then what does one pair test that the other doesn't?
* Add 9 and 25 to the test file, replacing 3 and 19 (implies churn in tracks that wish to keep exactly up to date with the test file)

I think it's best to make the README use 3 and 19. Coming right up.

**Comment by petertseng** [comment]
The commits for the canonical-data.json have been squashed down to one, and the README change added as a separate commit. I think the JSON change and the README change should be kept as separate commits? I don't know, I could be convinced either way on that one.

**Pull request merged**